doc/descriptors.md

```diff
@@ -0,0 +1,95 @@
+- `wsh(SCRIPT)` (not inside another 'wsh'): P2WSH embed the argument.
+- `pk(PUBKEY)` (anywhere): P2PK output for the given public key.
+- `pkh(PUBKEY)` (anywhere): P2PKH output for the given public key (use `addr` if you only thos the pubkey hash).
+- `wpkh(PUBKEY)` (not inside `wsh`): P2WPKH output for the given compressed pubkey.
+- `combo(PUBKEY)` (toplevel only): an alias for the collection of `pk(PUBKEY)`, `pkh(PUBKEY)`, `wpkh(PUBKEY)`, and `sh(wpkh(PUBKEY))`.
+- `multi(k,PUBKEY_1,PUBKEY_2,...,PUBKEY_n)` (anywhere): k-of-n multisig script.
+- `addr(ADDR)` (toplevel only): the script which ADDR expands to (note that there is no address for P2PK outputs).
+- `raw(HEX)` (toplevel only): the script whose hex encoding is HEX.
+
+Public keys:
+- Hex encoded public keys (66 characters starting with `02` or `03`, or 130 characters starting with `04`).
+  - Inside `wpkh` and `wsh`, only compressed public keys are permitted.
+- WIF encoded private keys, whose denote their corresponding public keys.
+- `xpub` encoded extended public key or `xprv` encoded private key
+  - Followed by zero or more `/NUM` unhardened and `/NUM'` hardened BIP32 derivation steps.
+  - Optionally followed by a single `/*` or `/*i` final step to denote all (direct) unhardened or hardened children.
+  - The usage of hardened derivation steps requires providing the private key.
+
+## Explanation
+
+### Single-key scripts
+
+Many single-key constructions are used in practice, generally including P2PK, P2PKH, P2WPKH, and P2SH-P2WPKH. Many more combinations are imaginable, though they may not be optimal: P2SH-P2PK, P2SH-P2PKH, P2WSH-P2PK, P2WSH-P2PKH, P2SH-P2WSH-P2PK, P2SH-P2WSH-P2PKH.
+
+To describe these, we model these as functions. The functions `pk` (P2PK), `pkh` (P2PKH) and `wpkh` (P2WPKH) take as input a public key in hexadecimal notation, and return the corresponding *scriptPubKey*. The functions `sh` (P2SH) and `wsh` (P2WSH) take as input a script, and return the script describing P2SH and P2WSH outputs with the input as embedded script. The names of the functions do not contain "p2" for brevity. Examples:
+* `pk(0279be667ef9dcbbac55a06295ce870b07029bfcdb2dce28d959f2815b16f81798)` represents a P2PK output.
+* `pkh(02c6047f9441ed7d6d3045406e95c07cd85c778e4b8cef3ca7abac09b95c709ee5)` represents a P2PKH output.
+* `wpkh(02f9308a019258c31049344f85f89d5229b531c845836f99b08601f113bce036f9)` represents a P2WPKH output.
+* `sh(wpkh(03fff97bd5755eeea420453a14355235d382f6472f8568a18b2f057a1460297556))` represents a P2SH-P2WPKH output.
+* `sh(wsh(pkh(02e493dbf1c10d80f3581e4904930b1404cc6c13900ee0758474fa94abe8c4cd13)))` represents a P2SH-P2WSH-P2PKH output.
+
+### Multisig
+
+Several pieces of software use multisignature scripts based on Bitcoin's OP_CHECKMULTISIG opcode. To support these, we introduce the `multi(k,key_1,key_2,...,key_n)` function. It represents a *k-of-n* multisig policy, where any *k* out of the *n* provided public keys must sign. Examples:
```

Comment:
Nit: "multisignature" should be "multi signature"?

Reply:
Done.